tox.ini

```diff
@@ -0,0 +1,16 @@
+[tox]
+envlist = py26, py27, py33, py34, py35, pypy
+
+[testenv]
+passenv = HYPOTHESIS_STORAGE_DIRECTORY
+
+deps =
+  nose-cov
+  hypothesis
+
+commands =
+  nosetests -v --logging-level=DEBUG --with-cov --cov chardet --cov-config .coveragerc
```

Comment:
Thanks for the explanation. You are right. That was because when using Travis called `tox -e py`, though the current Python version (i.e. PyPY) being used, the name of the environment will be `py` (instead `pypy`). Hence the override were ignored. I've updated the PR, to call `tox -e $(echo $TRAVIS_PYTHON_VERSION | sed 's/^\([0-9]\)\.\([0-9]\).*/py\1\2/')` instead, and the override seems to work, now. But this way we still don'T know to hard-code the Python versions redundantly.